packages/next-app/src/app/internalmanager/[network]/[operationKind]/[tokenAddress]/page.tsx

```diff
@@ -85,6 +88,22 @@ export default function Page({
     token: params.tokenAddress,
   });
 
+  useEffect(() => {
+    const tokenInfo = internalBalanceTokenData?.user
+      ?.userInternalBalances?.[0] || {
+      balance: "0",
+      tokenInfo: {
+        __typename: "Token",
+        address: params.tokenAddress,
+        decimals: walletAmount?.decimals || 0,
+        symbol: walletAmount?.symbol || "",
+        name: walletAmount?.symbol || "",
+      },
```

Comment:
could you say more to why this is needed here? Should we even be rendering this component if the user doesn't have any token balances?

Reply:
This is the case where the user doesn't have the token as an Internal Balance but has the token in their wallet, so the user can still do a deposit even with no internal balance (which is `balance`)